core/unit_test/TestTeamScan.hpp

```diff
@@ -186,10 +186,16 @@ struct TestTeamScanRetVal {
     a_r = view_2d_type("a_r", M, N);
     a_s = view_1d_type("a_s", M);
 
-    // Execute calculations
-    Kokkos::parallel_for(policy_type(M, Kokkos::AUTO), *this);
-
+    // Set team size explicitly to check whether non-power-of-two team sizes can
+    // be used.
+    if (ExecutionSpace().concurrency() > 10000)
+      Kokkos::parallel_for(policy_type(M, 127), *this);
+    else if (ExecutionSpace().concurrency() > 2)
+      Kokkos::parallel_for(policy_type(M, 3), *this);
+    else
+      Kokkos::parallel_for(policy_type(M, 1), *this);
```

Comment:
This is what's used on `develop` so we should probably cover all those cases here as well.

https://github.com/kokkos/kokkos/blob/d458fdadb0f0687cfa7579478450d60ed33c08cf/core/unit_test/TestTeamScan.hpp#L68-L75

Reply:
Looks like this doesn't work with the `OpenMP` implementation:
```
1: [ RUN      ] openmp.team_scan_ret_val
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 496
1:   a_os(i)
1:     Which is: 55
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 2016
1:   a_os(i)
1:     Which is: 231
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 8128
1:   a_os(i)
1:     Which is: 903
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 32640
1:   a_os(i)
1:     Which is: 3655
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 130816
1:   a_os(i)
1:     Which is: 14535
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 523776
1:   a_os(i)
1:     Which is: 58311
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:239: Failure
1: The difference between scan_ref and a_os(i) is 495, which exceeds abs_err, where
1: scan_ref evaluates to 561,
1: a_os(i) evaluates to 66, and
1: abs_err evaluates to 4.0531158447265625e-06.
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:239: Failure
1: The difference between scan_ref and a_os(i) is 1521, where
1: scan_ref evaluates to 1711,
1: a_os(i) evaluates to 190.
1: The abs_error parameter abs_err evaluates to 1.3100631690576847e-14 which is smaller than the minimum distance between doubles for numbers of this magnitude which is 2.8421709430404007e-14, thus making this EXPECT_NEAR check equivalent to EXPECT_EQUAL. Consider using EXPECT_DOUBLE_EQ instead.
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:239: Failure
1: The difference between scan_ref and a_os(i) is 1849, which exceeds abs_err, where
1: scan_ref evaluates to 2080,
1: a_os(i) evaluates to 231, and
1: abs_err evaluates to 7.7486038208007812e-06.
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:239: Failure
1: The difference between scan_ref and a_os(i) is 61009, where
1: scan_ref evaluates to 68635,
1: a_os(i) evaluates to 7626.
1: The abs_error parameter abs_err evaluates to 8.2378548427186615e-14 which is smaller than the minimum distance between doubles for numbers of this magnitude which is 9.0949470177292824e-13, thus making this EXPECT_NEAR check equivalent to EXPECT_EQUAL. Consider using EXPECT_DOUBLE_EQ instead.
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:237: Failure
1: Expected equality of these values:
1:   scan_ref
1:     Which is: 486591
1:   a_os(i)
1:     Which is: 53956
1: /var/jenkins/workspace/Kokkos/core/unit_test/TestTeamScan.hpp:239: Failure
1: The difference between scan_ref and a_os(i) is 763439, where
1: scan_ref evaluates to 858705,
1: a_os(i) evaluates to 95266.
1: The abs_error parameter abs_err evaluates to 2.9110047705671604e-13 which is smaller than the minimum distance between doubles for numbers of this magnitude which is 1.4551915228366852e-11, thus making this EXPECT_NEAR check equivalent to EXPECT_EQUAL. Consider using EXPECT_DOUBLE_EQ instead.
1: [  FAILED  ] openmp.team_scan_ret_val (147 ms)
```

Reply:
I don't understand this: the CI is passing, so where is this comment coming from?

Reply:
ok thanks!

Reply:
@cz4rs can you open an issue so that we don't forget about this ?

Reply:
Sure, #6470 opened.